### ISSUE TITLE
New cameras don't update

### ISSUE THREAD

**Issue by codypet** [twrecked/hass-aarlo#63]
I've used this for about a month now and its worked great.   Recently I changed one of the cameras to an Arlo Pro, and now I cannot get the HA interface to recognize the new camera much less get it so show up.   Is there something I need to do to get the new camera to show?  It shows up on the Arlo website fine and its viewable like the other cameras are.  I should note that the other Arlo Pro cameras on the same account are working just fine.

**Comment by twrecked** [comment]
Can you update and try again? I noticed an issue with the Arlo not sending me packets I need to determine devices. (Not sure the issue was mine or Arlo changing the back end.)

**Comment by codypet** [comment]
Should I be updating the whole HA instance or just replace the files from your repository?  I did restart HA and its still having the same problem.

**Comment by twrecked** [comment]
Just replace my files. It will go something like:
* `git pull` on the `aarlo` repository
* re-install the files (`./install go <where-you-put-stuff-your-config>`)
* stop home assistant
* start home assistant

And hopefully you'll see the new cameras.

**Comment by codypet** [comment]
Unfortunately, I'm still seeing just the old cameras.

**Comment by twrecked** [comment]
Crap. Are you ok to turn on some debug. I'll be able to see exactly what it thinks is happening. Add this to your configuration file:

```yaml
logger:
    default: info
    logs:
      custom_components.aarlo.alarm_control_panel: debug
      custom_components.aarlo.binary_sensor: debug
      custom_components.aarlo.camera: debug
      custom_components.aarlo.sensor: debug
      pyaarlo: debug
```

It will put some extra info into `config/home-assistant.log`.

**Comment by codypet** [comment]
Well I think I have my answer.   I moved a camera (Side yard) to another location (renaming it Back Yard East) and put a new camera in the old location(Naming the new camera Side yard).  Apparently Arlo never renamed the original side yard camera in the backend instead opting to name both Side yard.  Because the Aarlo component doesn't recognized the same name cameras, its not loading one of them.  
`2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/deviceId=48B14C7L0CEA9
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/deviceName=Side yard
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/deviceType=camera
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/mediaObjectCount=32510
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/parentId=4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/uniqueId=336-4496454_48B14C7L0CEA9
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/userId=4ANNY-336-13973168
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/presignedLastImageUrl=https://arlolastimage-z2.s3.amazonaws.co
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/presignedFullFrameSnapshotUrl=https://arlos3-prod-z2.s3.amazon
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7L0CEA9/xCloudId=FRHBNE-1005-336-44886832
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/deviceId=48B14C7Y08CD4
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/deviceName=Back Yard West
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/deviceType=camera
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/mediaObjectCount=5219
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/parentId=4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/uniqueId=336-4496454_48B14C7Y08CD4
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/userId=4ANNY-336-13973168
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/presignedLastImageUrl=https://arlolastimage-z2.s3.amazonaws.co
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/presignedFullFrameSnapshotUrl=https://arlos3-prod-z2.s3.amazon
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:48B14C7Y08CD4/xCloudId=FRHBNE-1005-336-44886832
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/deviceId=4XH16C7BC30FA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/deviceName=Front of House
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/deviceType=camera
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/mediaObjectCount=65780
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/parentId=4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/uniqueId=336-4496454_4XH16C7BC30FA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/userId=4ANNY-336-13973168
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/presignedLastImageUrl=https://arlolastimage-z2.s3.amazonaws.co
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/presignedFullFrameSnapshotUrl=https://arlos3-prod-z2.s3.amazon
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH16C7BC30FA/xCloudId=FRHBNE-1005-336-44886832
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/deviceId=4XH3747UAA101
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/deviceName=Side yard
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/deviceType=camera
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/mediaObjectCount=403
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/parentId=4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/uniqueId=336-4496454_4XH3747UAA101
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/userId=4ANNY-336-13973168
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/presignedLastImageUrl=https://arlolastimage-z2.s3.amazonaws.co
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/presignedFullFrameSnapshotUrl=https://arlos3-prod-z2.s3.amazon
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH3747UAA101/xCloudId=FRHBNE-1005-336-44886832
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/deviceId=4XH4757PAEA10
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/deviceName=Driveway
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/deviceType=camera
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/mediaObjectCount=70356
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/parentId=4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/uniqueId=336-4496454_4XH4757PAEA10
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/userId=4ANNY-336-13973168
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/presignedLastImageUrl=https://arlolastimage-z2.s3.amazonaws.co
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/presignedFullFrameSnapshotUrl=https://arlos3-prod-z2.s3.amazon
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4XH4757PAEA10/xCloudId=FRHBNE-1005-336-44886832
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/deviceId=4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/deviceName=Lake Como
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/deviceType=basestation
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/mediaObjectCount=0
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/uniqueId=336-4496454_4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/userId=4ANNY-336-13973168
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:4892547AA1EFA/xCloudId=FRHBNE-1005-336-44886832
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:ARLO/totalCameras=5
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] set:ARLO/totalDoorBells=0
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] getting initial settings
2019-05-24 00:50:46 DEBUG (ArloBackgroundWorker) [pyaarlo] starting request=https://arlo.netgear.com/hmsweb/users/devices/notify/4892547AA1EFA
2019-05-24 00:50:46 DEBUG (Thread-11) [pyaarlo] registering cron jobs
2019-05-24 00:50:46 INFO (MainThread) [homeassistant.setup] Setup of domain aarlo took 7.4 seconds.
2019-05-24 00:50:46 INFO (MainThread) [homeassistant.components.camera] Setting up camera.aarlo
2019-05-24 00:50:46 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.aarlo
2019-05-24 00:50:46 INFO (MainThread) [custom_components.aarlo.camera] ArloCam: Side yard created
2019-05-24 00:50:46 INFO (MainThread) [custom_components.aarlo.camera] ArloCam: Back Yard West created
2019-05-24 00:50:46 INFO (MainThread) [custom_components.aarlo.camera] ArloCam: Front of House created
2019-05-24 00:50:46 INFO (MainThread) [custom_components.aarlo.camera] ArloCam: Side yard created
2019-05-24 00:50:46 INFO (MainThread) [custom_components.aarlo.camera] ArloCam: Driveway created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Captured Today Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Captured Today Back Yard West created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Captured Today Front of House created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Captured Today Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Captured Today Driveway created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Last Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Last Back Yard West created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Last Front of House created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Last Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Last Driveway created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Arlo Cameras created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Battery Level Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Battery Level Back Yard West created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Battery Level Front of House created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Battery Level Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Battery Level Driveway created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Signal Strength Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Signal Strength Back Yard West created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Signal Strength Front of House created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Signal Strength Side yard created
2019-05-24 00:50:47 INFO (MainThread) [custom_components.aarlo.sensor] ArloSensor: Signal Strength Driveway created
2019-05-24 00:50:47 INFO (MainThread) [homeassistant.setup] Setup of domain light took 3.5 seconds.
2019-05-24 00:50:47 INFO (MainThread) [homeassistant.components.light] Setting up light.zwave
2019-05-24 00:50:47 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/entity_platform.py", line 357, in _async_add_entity
    raise HomeAssistantError(msg)
homeassistant.exceptions.HomeAssistantError: Entity id already exists: camera.aarlo_side_yard. Platform aarlo does not generate unique IDs
2019-05-24 00:50:47 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/entity_platform.py", line 357, in _async_add_entity
    raise HomeAssistantError(msg)
homeassistant.exceptions.HomeAssistantError: Entity id already exists: sensor.aarlo_last_side_yard. Platform aarlo does not generate unique IDs
2019-05-24 00:50:47 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/entity_platform.py", line 357, in _async_add_entity
    raise HomeAssistantError(msg)
homeassistant.exceptions.HomeAssistantError: Entity id already exists: sensor.aarlo_captured_today_side_yard. Platform aarlo does not generate unique IDs
2019-05-24 00:50:47 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/entity_platform.py", line 357, in _async_add_entity
    raise HomeAssistantError(msg)
homeassistant.exceptions.HomeAssistantError: Entity id already exists: sensor.aarlo_battery_level_side_yard. Platform aarlo does not generate unique IDs
2019-05-24 00:50:47 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/entity_platform.py", line 357, in _async_add_entity
    raise HomeAssistantError(msg)
homeassistant.exceptions.HomeAssistantError: Entity id already exists: sensor.aarlo_signal_strength_side_yard. Platform aarlo does not generate unique IDs
2019-05-24 00:50:47 INFO (MainThread) [homeassistant.setup] Setup of domain ring took 7.9 seconds.
2019-05-24 00:50:47 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.ring
2019-05-24 00:50:47 INFO (MainThread) [homeassistant.components.camera] Setting up camera.ring
2019-05-24 00:50:47 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.ring
2019-05-24 00:50:47 DEBUG (EventStream) [pyaarlo] async ping response subscriptions/4ANNY-336-13973168_web
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing image update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing media update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing image update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing media update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing image update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing media update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing image update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing media update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing image update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] queing media update
2019-05-24 00:50:48 DEBUG (ArloBackgroundWorker) [pyaarlo] getting image for Side yard`
Any idea on how to get Arlo to rename the camera on the backend?

**Comment by twrecked** [comment]
Are you willing to check out a branch to try. I think I can fix this at the Home Assistant end.

**Comment by twrecked** [comment]
This https://github.com/twrecked/hass-aarlo/tree/device-timestamp passes a timestamp into the devices request.

**Comment by codypet** [comment]
It appears that was the problem.  I'll let you know if it has any problems, but at first glance everything has come back up.  I'm rebooting now for an unrelated reason so I'll let you know if its still working.

**Comment by twrecked** [comment]
Excellent. I'll merge that over to master some time today.

**Comment by twrecked** [comment]
Merged with #65
